sr_grasp/src/sr_grasp/__init__.py

```diff
@@ -0,0 +1,111 @@
+
+from copy import deepcopy
+import os, yaml
+from rospy import logerr, loginfo
+import rospkg, genpy
+import moveit_msgs.msg
+from sr_grasp.utils import mk_grasp
+
+class Grasp(moveit_msgs.msg.Grasp):
+    """
+    Represents a single grasp, basically a wrapper around moveit_msgs/Grasp.
+    """
+    def __init__(self):
+        super(Grasp, self).__init__()
+
+    @classmethod
+    def from_msg(cls, msg):
+        """Construct a shadow grasp object from moveit grasp object."""
+        grasp = Grasp()
+        grasp.id = msg.id
+        grasp.pre_grasp_posture  = deepcopy(msg.pre_grasp_posture)
+        grasp.grasp_posture      = deepcopy(msg.grasp_posture)
+        grasp.grasp_pose         = deepcopy(msg.grasp_pose)
+        grasp.grasp_quality      = msg.grasp_quality
+        grasp.pre_grasp_approach = deepcopy(msg.pre_grasp_approach)
+        grasp.post_grasp_retreat = deepcopy(msg.post_grasp_retreat)
+        grasp.post_place_retreat = deepcopy(msg.post_place_retreat)
+        grasp.max_contact_force  = msg.max_contact_force
+        grasp.allowed_touch_objects = deepcopy(msg.allowed_touch_objects)
+        return grasp
+
+    def to_msg(self):
+        """Return plain moveit_msgs/Grasp version of self."""
+        raise Exception("TODO - to_msg")
+
+    @classmethod
+    def from_yaml(self, y):
+        """
+        Construct a shadow grasp object from YAML object. For example YAML
+        grabbed from rostopic to a file.
+        """
+        grasp = Grasp()
+        genpy.message.fill_message_args(grasp, y)
+        return grasp
+
+
+# Store of loaded grasps. Global var as we want multiple instances of the class
+# within a process to share the same data. IE that act like clients. This will
+# make more sense when the grasp shash becomes a proper node with databases and
+# the like.
+_store = {}
```

Comment:
this sounds scary. Are you 100% sure we want that behaviour? (instead of sharing the same instance of a GraspStash object, or pointing to the same parameter)?